services/parser-database/parser.py

```diff
@@ -0,0 +1,218 @@
+    "loads/2019/09/Nuevo-Reglamento-Homologado-1.pdf",
+}
+document_list = []
+document_list.append(mty_document)
+
+
+def parse_all_documents():
+    """Parses all documents that are specified on the DB
+    (For obvious reasons, this won't work as is while the DB is not in use)
+    (But will still parse the hardcoded document)"""
+    for document in document_list:
+        parse(document)
+
+
+def has_file_changed(past_hash):
+    """Sees if the file is different.
+
+    Args:
+        past_hash (string): hash to compare from
+
+    Returns:
+        [boolean]: [if the file has changed or not]
+    """
+    hasher = hashlib.sha256()
+    with open("regs.pdf", "rb") as pdf_file:
+        file_buffer = pdf_file.read()
+        hasher.update(file_buffer)
+        if hasher.hexdigest() == past_hash:
+            return False
+        return True
+
+
+# TODO Change to point to Javier's service
+def get_keywords(text):
+    """Get keywords that relate to this article (Javier's service)
+
+    Args:
+        text (sting): text to extract keywords from
+
+    Returns:
+        [JSON]: extracted keywords
+    """
+    """
+    return requests.post(
+        "localhost:8000", params={"text": text}
+    ).json()
+    """
+    keywords_service_response = keywordmock.get_keywords(text)
+    return json.loads(keywords_service_response)
+
+
+def article_to_dictionary(article):
+    """Converts an parsed article to a dictionary and saves it
+
+    Args:
+        article (Article): [article object]
+    """
+    article_dict = {
+        "articleNumber": article.number,
+        "text": article.text,
+        "wordCount": len(article.text.split())
+    }
+    articles_in_memory.append(article_dict)
+    save_keywords_in_memory(get_keywords(article.text), article)
+
+
+def save_keywords_in_memory(keywords, article):
+    """Saves the keywords from an article in memory
+
+    Args:
+        keywords (JSON): contains keywords
+        article (Article): article object
+    """
+    split_article = article.text.split()
+    for keyword in keywords["keywords"]:
+        frequency = split_article.count(keyword)
+        if keyword not in keywords:
+            keywords_in_memory[keyword] = []
+        keywords_in_memory[keyword].append({
+            "articleNumber": article.number,
+            "frequency": frequency
+        })
+
+
+# When the time comes to implement the DB, the commented stuff will allow
+# that (with a few additions)
+def parse(document_to_parse):
+    """Parses all PDF documents that are in the DB"""
+    retriever.get_document(document_to_parse["url"])
+    print('File downloaded')
+    if has_file_changed(document_to_parse["hash"]):
+        print('File has changed')
+        with open("regs.pdf", "rb") as pdf_file:
+            doc = slate.PDF(pdf_file)
+            final_text = ""
+            for page in doc:
+                final_text += page
+            final_text = final_text.strip().lower().split()
```

Comment:
You probably don't want to lower() here, because capitalization is an important part of grammar and the text will eventually get shown to end users.

Reply:
will see how I can adapt my algorithm, with the regex the lower() wouldn't be necessary so definitely something to look into